CMakeLists.txt

```diff
@@ -440,25 +449,39 @@ add_subdirectory(${FLB_PATH_LIB_CMETRICS} EXCLUDE_FROM_ALL)
 add_subdirectory(${FLB_PATH_LIB_CTRACES} EXCLUDE_FROM_ALL)
 
 # Nghttp2 options
-FLB_OPTION(ENABLE_LIB_ONLY   ON)
-FLB_OPTION(ENABLE_STATIC_LIB ON)
-FLB_OPTION(ENABLE_SHARED_LIB OFF)
-
-FLB_DEFINITION(NGHTTP2_STATICLIB)
-
-add_subdirectory(${FLB_PATH_LIB_NGHTTP2} EXCLUDE_FROM_ALL)
+if(FLB_PREFER_SYSTEM_LIB_NGHTTP2)
+  find_package(PkgConfig)
+  pkg_check_modules(NGHTTP2 libnghttp2>=1.0.0)
+endif()
+if(NGHTTP2_FOUND)
+  include_directories(${NGHTTP2_INCLUDE_DIRS})
+  link_directories(${NGHTTP2_LIBRARY_DIRS})
+else()
+  include(cmake/nghttp2.cmake)
+endif()
 
 # C-Ares (DNS library)
-FLB_OPTION(CARES_STATIC      ON)
-FLB_OPTION(CARES_SHARED      OFF)
-FLB_OPTION(CARES_INSTALL     OFF)
-FLB_OPTION(CARES_BUILD_TESTS OFF)
-FLB_OPTION(CARES_BUILD_TOOLS OFF)
-if (FLB_SYSTEM_MACOS)
-  # macOS SDK always has <arpa/nameser.h>.
-  FLB_DEFINITION(CARES_HAVE_ARPA_NAMESER_H)
+if(FLB_PREFER_SYSTEM_LIB_CARES)
+  find_package(PkgConfig)
+  pkg_check_modules(CARES libcares>=1.0.0)
+endif()
+if(CARES_FOUND)
+  include_directories(${CARES_INCLUDE_DIRS})
+  link_directories(${CARES_LIBRARY_DIRS})
+else()
+  FLB_OPTION(CARES_STATIC      ON)
+  FLB_OPTION(CARES_SHARED      OFF)
+  FLB_OPTION(CARES_INSTALL     OFF)
+  FLB_OPTION(CARES_BUILD_TESTS OFF)
+  FLB_OPTION(CARES_BUILD_TOOLS OFF)
+  if (FLB_SYSTEM_MACOS)
+    # macOS SDK always has <arpa/nameser.h>.
+    FLB_DEFINITION(CARES_HAVE_ARPA_NAMESER_H)
+  endif()
+  add_subdirectory(${FLB_PATH_LIB_CARES})# EXCLUDE_FROM_ALL)
+
```

Comment:
I believe that CARES' headers is also included in cmake/headers.cmake.
So, we need to remove there and need to create cares.cmake like as kafka, nghttp2, etc.

Reply:
I've updated the PR.